crates/fuel-gas-price-algorithm/src/v1/tests/update_da_record_data_tests.rs

```diff
@@ -270,3 +228,132 @@ fn update_da_record_data__da_block_updates_projected_total_cost_with_known_and_g
     let expected = new_known_total_cost + guessed_part;
     assert_eq!(actual, expected as u128);
 }
+
+#[test]
+fn update_da_record_data__updates_known_total_cost_if_blocks_are_out_of_order() {
+    // given
+    let da_cost_per_byte = 20;
+    let block_bytes = 1000;
+    let unrecorded_blocks = vec![
+        BlockBytes {
+            height: 1,
+            block_bytes,
+        },
+        BlockBytes {
+            height: 2,
+            block_bytes,
+        },
+        BlockBytes {
+            height: 3,
+            block_bytes,
+        },
+    ];
+    let old_known_total_cost = 500;
+    let old_projected_total_cost =
+        old_known_total_cost + (block_bytes as u128 * da_cost_per_byte * 3);
+    let old_da_cost_per_byte = 20;
+    let mut updater = UpdaterBuilder::new()
+        .with_da_cost_per_byte(da_cost_per_byte)
+        .with_unrecorded_blocks(unrecorded_blocks)
+        .with_da_cost_per_byte(old_da_cost_per_byte)
+        .with_known_total_cost(old_known_total_cost)
+        .with_projected_total_cost(old_projected_total_cost)
+        .build();
+    let new_cost_per_byte = 100;
+    let recorded_cost = 2 * (block_bytes * new_cost_per_byte) as u128;
+    let recorded_range: Vec<u32> = vec![2, 3];
+
+    // when
+    updater
+        .update_da_record_data(&recorded_range, recorded_cost)
+        .unwrap();
+
+    // and
+    let expected = updater.latest_known_total_da_cost_excess
+        + (block_bytes * new_cost_per_byte) as u128;
+    let actual = updater.projected_total_da_cost;
+    assert_eq!(actual, expected);
+}
+#[test]
+fn update_da_record_data__updates_projected_total_cost_if_blocks_are_out_of_order() {
+    // given
+    let da_cost_per_byte = 20;
+    let block_bytes = 1000;
+    let unrecorded_blocks = vec![
+        BlockBytes {
+            height: 1,
+            block_bytes,
+        },
+        BlockBytes {
+            height: 2,
+            block_bytes,
+        },
+        BlockBytes {
+            height: 3,
+            block_bytes,
+        },
+    ];
+    let old_known_total_cost = 500;
+    let old_projected_total_cost =
+        old_known_total_cost + (block_bytes as u128 * da_cost_per_byte * 3);
+    let old_da_cost_per_byte = 20;
+    let mut updater = UpdaterBuilder::new()
+        .with_da_cost_per_byte(da_cost_per_byte)
+        .with_unrecorded_blocks(unrecorded_blocks)
+        .with_da_cost_per_byte(old_da_cost_per_byte)
+        .with_known_total_cost(old_known_total_cost)
+        .with_projected_total_cost(old_projected_total_cost)
+        .build();
+    let new_cost_per_byte = 100;
+    let recorded_cost = 2 * (block_bytes * new_cost_per_byte) as u128;
+    let recorded_range: Vec<u32> = vec![2, 3];
```

Comment:
Nit: Not sure if we should call this a range if it's a vec.

```suggestion
    let recorded_heights: Vec<u32> = vec![2, 3];
```